app/js/directives.js

```diff
@@ -1566,6 +1570,85 @@ angular.module('myApp.directives', ['myApp.filters'])
         });
       });
 
+      if (Config.Navigator.ffos) {
+
+        navigator.getUserMedia = ( navigator.getUserMedia || navigator.webkitGetUserMedia || navigator.mozGetUserMedia || navigator.msGetUserMedia);
+
+        voiceRecord.on('touchstart', function(e) {
+          if ($scope.$parent.$parent.voiceRecorder.processing) { return; }
+
+          navigator.getUserMedia({audio : true}, function(stream){
+            var start = Date.now();
+            var touch = null;
+
+            audioPromise = null;
+            audioStream = stream;
+            audioRecorder = new MediaRecorder(stream);
+
+            var interval = setInterval(function(){
+              var time = (new Date());
+
+              time.setTime(Date.now() - start);
+
+              $scope.$apply(function(){
+                $scope.$parent.$parent.voiceRecorder.time = (time.getMinutes() < 10 ? '0' : '') + time.getMinutes() + ':' + (time.getSeconds() < 10 ? '0' : '') + time.getSeconds();
+              });
+            }, 1000);
+
+            $scope.$apply(function(){
+              $scope.$parent.$parent.voiceRecorder.time = '00:00';
+              $scope.$parent.$parent.voiceRecorder.recording = interval;
+            });
+
+            audioRecorder.start();
+
+            console.log('recording now!');
+
+          }, function(e){
+            console.error(e);
+          });
+        });
+
+        voiceRecord.on('click', function(){
+          if (audioPromise) {
```

Comment:
I only saw it setting to a value different than `null` below. Are you sure, this code path is ever entered?